[Estuary] Video Versions Dialog Redesign

jjd-uk opened the pull request:
## Description
Modifiy the dialogs used for the video versions and extras

## Motivation and context
Improve the presented info to allow choice of version/extra

Note also several variables have been introduced to help with the media info provided. These variables will not only be for these dialogs as I have a media flag rework which I will PR that will also make use of these variables.

## How has this been tested?

Has been locally tested

## What is the effect on users?
Improve the look of the video versions feature and provide useful media info in deciding between versions.

## Screenshots (if appropriate):

**Before Select dialog**

![image](https://github.com/xbmc/xbmc/assets/5781142/da11e3ce-840d-4815-8775-53cdbf28fbf2)

**After Select dialog**

![image](https://github.com/xbmc/xbmc/assets/5781142/c6bb3740-21aa-4bac-90ea-5ccd42a4c0ce)

**Before Manage dialog**

![image](https://github.com/xbmc/xbmc/assets/5781142/e84567a6-0dd4-47eb-8dc7-4306b78dcef5)

![image](https://github.com/xbmc/xbmc/assets/5781142/6615ca02-d455-4ac3-b969-8c782ea56c9a)

**After Manage dialog**

![image](https://github.com/xbmc/xbmc/assets/5781142/b1d53750-e29f-41f6-8f7f-8fd5da58ae0f)

![image](https://github.com/xbmc/xbmc/assets/5781142/8b6c3807-d81b-4ac8-9016-37e81dfcdce8)


## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

jjd-uk commented:
Please note that I'm not going to have any access to a computer for the next week, so the earliest I'll be able to look at any requested changes will the 29/12/23.

ksooo commented:
@jjd-uk  Are the fake context menu item entries generated dynamically, for example "remove" and "set default" not being available if a default video version is selected?

jjd-uk commented:
@ksooo what is listed in the fake context is dynamic and is based on what controls are enabled when selecting an item. What is enabled on selection is controlled by the c++ side of things, on the skin side we simply rely on the c++ setting focus to the play button on selection of an item in the list.

jjd-uk commented:
If default version is selected

![image](https://github.com/xbmc/xbmc/assets/5781142/3f15bea7-80ce-4d8e-98e5-4cef2cef304f)

CrystalP commented:
Comments as a user as I don't know skinning.

1. Choose version dialog: I like that the duration and media flags appear where the filename used to be. The dialog works well with the mouse. However I miss the filename - maybe as a line of text at the bottom of the dialog, like in Manage?

2. The rest of the comments is for the Manage screen.
* Mouse navigation works well on the Manage dialog itself, but not on the actions popup. 

If the position of the mouse when clicking on a version is not already where the popup appears, moving the mouse to the popup dismisses it. That doesn't happen for the context menu / sub-context menu. See video:

https://github.com/xbmc/xbmc/assets/489377/3155960e-9b58-4910-91b4-3018050c37f1

* There is no visual indication of which version is the default. Is something needed from core? What?

* The filename at the bottom of the dialog is not updated when moving up/down between versions

* At first I didn't see the "Press OK to open menu" text (was against a relatively clear/bright background).

* probably low priority, on keyboard I had to look for the way to dismiss the popup without closing the Manage dialog at the same time (Backspace key fits the bill, Escape closes both) and it wasn't mapped on my remote, but I have a non-standard setup, don't know what happens with standard mappings.

* A couple things about "Play" action in the popup, any change would likely be in C++:
  * Issue: play on a specific version goes to the "Choose version" dialog to choose the version again, instead of playing the version that opened the popup.
  * What is its purpose? After playback ends you're back in the library, not in the Manage screen anymore, so you can't use that Play button to try different files or edit name/artwork without losing your place.

* Maybe requires core support: the default version can't be removed? Yes the default version is "special" but removal should be possible with an automated change of the default version followed by removal of the former default.

jurialmunkey commented:
@CrystalP 

> There is no visual indication of which version is the default. Is something needed from core? What?

The IsSelected boolean needs to be set for the ListItem, then the selected highlight colour will automatically be used for the label.

We were waiting on a fix for that before.

> probably low priority, on keyboard I had to look for the way to dismiss the popup without closing the Manage dialog at the same time (Backspace key fits the bill, Escape closes both) and it wasn't mapped on my remote, but I have a non-standard setup, don't know what happens with standard mappings.

Skin only has control over the onback behaviour.

In standard keymap, backspace is mapped to `Action(Back)` whereas escape is mapped to `Action(PreviousMenu)` hence the different behaviour. 

End goal would be to remove this custom dialog entirely as all the behaviour is achievable via native dialogs (DialogSelect and ContextMenu) and having a custom dialog for this purpose is excessive -- but that is dependent on someone knowledgeable doing the work, so for now we bodge it via skin code to make it somewhat usable in the interim.

On a standard remote, back should be back action (it is on mine) so it should be okay for now with a default setup.

> Play action

This imho should be removed entirely from the management dialog as it is not a management function. 
I expect this would happen as part of the switch to native dialogs.


> Maybe requires core support: the default version can't be removed?

Definitely needs core support. The actions of all the buttons are hard-coded and skin can't control their behaviour (only appearance).

I can't see a reason why it wouldn't be possible though (assuming there is more than one version available). It'd simply be a case of setting the default to the next available version, then removing the entry for the old one.

jurialmunkey commented on addons/skin.estuary/xml/DialogVideoVersion.xml:
```diff
@@ -1,167 +1,399 @@
+				<align>right</align>
+				<aligny>bottom</aligny>
+				<textcolor>grey</textcolor>
+				<label>$INFO[ListItem.FileNameAndPath]</label>
```
Use a $VAR here to get `Container(ID).ListItem.FileNameAndPath` for active list (see @CrystalP comments)

i.e.
```
<variable name="VideoVersionsFileNameAndPath">
    <value condition="String.IsEqual(window(home).Property(dialogvideoversion),extras)">$INFO[Container(51).ListItem.FileNameAndPath]</value>
    <value>$INFO[Container(50).ListItem.FileNameAndPath]</value>
</variable>
```

HitcherUK commented:
> 1. Choose version dialog: I like that the duration and media flags appear where the filename used to be. The dialog works well with the mouse. However I miss the filename - maybe as a line of text at the bottom of the dialog, like in Manage?

There's really no need to know the file name once you've set up your versions as they tell you which version they are.

jjd-uk commented:
> * The filename at the bottom of the dialog is not updated when moving up/down between versions

Updated as I had some time before I start my Xmas travels. This is now fixed.

> *Mouse navigation works well on the Manage dialog itself, but not on the actions popup.

Ok I missed that, it's the one part I did not check with a mouse, not sure if we can do anything about it.

Any ideas @jurialmunkey 

It was introduced to make it clearer on what item in list the action was to be performed on. If we can't solve the issue with mouse navigation then we can go back to the buttons being on the right hand side.

![image](https://github.com/xbmc/xbmc/assets/5781142/428fa169-54ab-4961-a76a-c59566192b22)

jjd-uk commented:
> > 1. Choose version dialog: I like that the duration and media flags appear where the filename used to be. The dialog works well with the mouse. However I miss the filename - maybe as a line of text at the bottom of the dialog, like in Manage?
> 
> There's really no need to know the file name once you've set up your versions as they tell you which version they are.

I did consider having the file path in both the Choose and Manage dialogs, but we don't normally show file paths for library items hence it seemed not necessary to me in the Choose dialog. Easy enough to add however if there's a consensus for that.

jjd-uk commented:
> * At first I didn't see the "Press OK to open menu" text (was against a relatively clear/bright background).

I don't understand this one. When the dialog is opened the background behind the dialog is darken so there should be anything bright in the background.

CrystalP commented:
> > > 1. Choose version dialog: I like that the duration and media flags appear where the filename used to be. The dialog works well with the mouse. However I miss the filename - maybe as a line of text at the bottom of the dialog, like in Manage?
> > 
> > 
> > There's really no need to know the file name once you've set up your versions as they tell you which version they are.
> 
> I did consider having the file path in both the Choose and Manage dialogs, but we don't normally show file paths for library items hence it seemed not necessary to me in the Choose dialog. Easy enough to add however if there's a consensus for that.

No big deal, I don't have a good "regular user" use case. It's more useful for dev. Having it in Manage works.


> > *Mouse navigation works well on the Manage dialog itself, but not on the actions popup.
> 
> Ok I missed that, it's the one part I did not check with a mouse, not sure if we can do anything about it.

I like the idea of that popup, do you need core support to make it a "real" popup menu, like the context menu?





> > There is no visual indication of which version is the default. Is something needed from core? What?
> 
> The IsSelected boolean needs to be set for the ListItem, then the selected highlight colour will automatically be used for the label.

I can take a shot at this. What I remember from earlier was setting a listitem properly to move an item to the of of the list.



> End goal would be to remove this custom dialog entirely as all the behaviour is achievable via native dialogs (DialogSelect and ContextMenu) and having a custom dialog for this purpose is excessive -- but that is dependent on someone knowledgeable doing the work, so for now we bodge it via skin code to make it somewhat usable in the interim.

I can take a look at the core side if there is a consensus on the look & feel, the Manage dialog after this PR looks pretty similar to the Choose version, I would use that as inspiration.



> > * At first I didn't see the "Press OK to open menu" text (was against a relatively clear/bright background).
> 
> I don't understand this one. When the dialog is opened the background behind the dialog is darkened so there should be anything bright in the background.

See the video I included in first comment, you can see it.

> > Play action
> 
> This imho should be removed entirely from the management dialog as it is not a management function. I expect this would happen as part of the switch to native dialogs.

Would be included in an attempt to rework as select dialog, if there is a consensus for removal. Even without a refactor it shouldn't be difficult to disable it from core.



> > Maybe requires core support: the default version can't be removed?
> 
> Definitely needs core support. The actions of all the buttons are hard-coded and skin can't control their behaviour (only appearance).
> 
> I can't see a reason why it wouldn't be possible though (assuming there is more than one version available). It'd simply be a case of setting the default to the next available version, then removing the entry for the old one.

I'll take a crack while I'm at it.

Sorry for the essay!

jjd-uk commented:
@CrystalP ok have watched the video, doesn't seem to have darken as much as work have expected. I'm without computer access for the next week now to check it for myself. That's trival to change however if there's anyway we can get this fake context pop up to work with a mouse. If not then it's academic anyway as we'll have to switch back to the side buttons.

Best solution would be a real context menu, but I have zero idea it's possible to have a context menu for dialog items, or how much work it would be if it was possible.

ksooo commented:
Maybe for Omega we stick with the simple approach to just use simple buttons, no fake context menu or native context menu? The latter is possible, but has definitely no priority for Omega.

HitcherUK commented:
> Maybe for Omega we stick with the simple approach to just use side buttons, no fake context menu, no native context menu? The latter is possible, but has definitely no priority for Omega. The first seems to be more a hack than a clean solution.

Agreed, we don't want to confuse users with a method that looks like a context menu, but doesn't open using the context button, that eventually can only be opened using the context button.

jjd-uk commented:
Ok will update with all action buttons on the right once I'm back from my travels.

jurialmunkey commented:
The problem with the side buttons is that the buttons work for the item you click, not the one in focus. It is even more confusing (and worse with the mouse - at least with remote/keys a select action can be forced onright). 

With side buttons, a user will think they're editing the focused item but they're actually editing a random item in the list with no indication of which one it is. That's far worse imho. 

Neither is a good approach - the fake context is making do with a badly designed dialog to make it somewhat usable by forcing a click so the user is certain what they're working on.

Imho going back to side buttons requires the focus issue to be fixed in core first.

Alternatively, I do have an idea which potentially will address the keeping the fake context active for mouse but obviously cannot do anything about using the context button itself.

jurialmunkey commented:
This patch will fix the mouse focus issue. It also fixes the animation of the context menu pop-up.

https://gist.github.com/jurialmunkey/b418ebc6ee9f24dce74478a45747b625#file-videoversionsmousefix-patch

Clicking with mouse anywhere outside the context menu should now close it (same behaviour as "real" context menu). Context will not lose focus if mouse cursor is not inside

CrystalP commented:
Tried it, it works. Found one problem though: Manage Version > click on movie > click play. end playback. Go back to manage versions > the popup is already open. All mouse navigation, using keyboard for some parts doesn't make it happen, I didn't dig for details.

Agree with your arguments for a popup version rather than buttons on the side, they made the original dialog confusing.

jurialmunkey commented:
@CrystalP - Ah I see what you mean. That's a weird one. 

I made a revision to the patch to make the onclick behaviour more similar to standard contextmenu (i.e. close context before performing the selected action). That also inadvertently fixes that play button issue because focus is sent back to the list first before performing the action.

https://gist.github.com/jurialmunkey/b418ebc6ee9f24dce74478a45747b625#file-videoversionsmousefix-patch

CrystalP commented:
That worked all but one time, good enough for me.
Found another weird one, but I'd say a low priority.
If you're moving the mouse (any direction) as you click on Set Default, the action is executed and the popup immediately pops back up, or sometimes just remains visible, minus one button.

CrystalP commented:
One more remark about the manage screen default buttons: the back button should probably be labelled "Close" instead of "Cancel", since all changes are saved immediately and will not be cancelled by closing the screen.
Maybe the message 15067?

jurialmunkey commented:
>  If you're moving the mouse (any direction) as you click on Set Default, the action is executed and the popup immediately pops back up, or sometimes just remains visible, minus one button.

This will be the fade out on menu/background because technically the buttons stay in a "visible" state until the "hidden" animation state completes (rather than starts). Leaves about 200ms where you can bump the mouse to get the button to regain focus.

The "fix" is removing the hidden animation so it hides immediately. It's a trade-off with some slightly janky behaviour either way - just comes down to which is more important I guess.

> the back button should probably be labelled "Close" instead of "Cancel"

Good point. Definitely agree.

jjd-uk commented:
@Hitcher @jurialmunkey @CrystalP @ksooo Rebased onto the changes in Master and rewritten to take account of current state versions/extras in Master.

All comments related to previous state of PR have been hidden just to keep conversation history clean and relevent to current state of PR. They can still be shown if anyone wants to go back over them.

jjd-uk commented:
The one issue I should mention.

`DialogSelect` sets `defaultcontrol` to be id `3`

This means I have to used id `3 `in my new layout `VideoSelectLayout` in `Includes_DialogSelect` although it doesn't do anything as it just an empty control, as removing it breaks the layout. One possible solution is to set `defaultcontrol` to be id `6` in `DialogSelect`, and while in testing it fixed the issue and I couldn't find any adverse elsewhere, I wasn't sure about taking that risk at this time. So for now  `defaultcontrol` has been left as id `3`.

jurialmunkey commented:
@jjd-uk I think you should be able to solve the defaultcontrol issue with a conditional include

i.e. somewhere in an Includes file define an include:
```xml
<include name="DefaultControl">
    <defaultcontrol always="$PARAM[always]">$PARAM[id]</defaultcontrol>
</include>
```

Then in DialogSelect replace the `<defaultcontrol>` tags with:
```
<include content="DefaultControl" condition="Window.IsActive(selectvideoversion) | Window.IsActive(selectvideoextra)">
    <param name="always">true</param>
    <param name="id">6</param>
</include>
<include content="DefaultControl" condition="![Window.IsActive(selectvideoversion) | Window.IsActive(selectvideoextra)]">
    <param name="always">true</param>
    <param name="id">3</param>
</include>
```


That's normally what I do for this type of scenario, so I think it should work but I haven't tested specifically with videoversions in DialogSelect.

The alternative if that doesn't work is a conditional onload setfocus. Not as much a fan of onload setfocus as it doesn't return focus to the control if it gets lost (e.g. from mouse moving). Much prefer defaultcontrol.
```xml
<onload condition="Window.IsActive(selectvideoversion) | Window.IsActive(selectvideoextra)">SetFocus(6)</onload>
```

jurialmunkey commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -676,12 +677,109 @@
+		<value condition="!String.IsEmpty(container(6).listitem.art(poster))">$INFO[container(6).listitem.art(poster)]</value>
+		<value condition="!String.IsEmpty(container(6).listitem.art(thumb))">$INFO[container(6).listitem.art(thumb)]</value>
+		<value condition="!String.IsEmpty(container(50).listitem.art(poster)) + Control.IsVisible(50)">$INFO[container(50).listitem.art(poster)]</value>
+		<value condition="!String.IsEmpty(container(50).listitem.art(thumb))+ Control.IsVisible(50)">$INFO[container(50).listitem.art(thumb)]</value>
```
Not sure if it actually matters in terms of function but you missed a space here before the `+`

jjd-uk replied:
I don't think those Control.IsVisible conditions are required anymore, they are a legacy from the dual lists in the same dialog, so will confirm they can be removed.

jurialmunkey commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -676,12 +677,109 @@
+		<value condition="!String.IsEmpty(container(6).listitem.art(thumb))">$INFO[container(6).listitem.art(thumb)]</value>
+		<value condition="!String.IsEmpty(container(50).listitem.art(poster)) + Control.IsVisible(50)">$INFO[container(50).listitem.art(poster)]</value>
+		<value condition="!String.IsEmpty(container(50).listitem.art(thumb))+ Control.IsVisible(50)">$INFO[container(50).listitem.art(thumb)]</value>
+		<value condition="String.IsEmpty(Listitem.Art(poster)) ">$INFO[Listitem.Art(poster)]</value>
```
Super minor, but missed some trailing whitespace in the condition.

jurialmunkey commented on addons/skin.estuary/xml/DialogSelect.xml:
```diff
@@ -4,11 +4,12 @@
 		<include condition="Window.IsActive(gamestretchmode)">GameDialogSelectViewLayout</include>
 		<include condition="Window.IsActive(gamevideorotation)">GameDialogSelectViewLayout</include>
 		<include condition="Window.IsActive(ingamesaves)">GameDialogSelectSaveInGameLayout</include>
+		<include condition="Window.IsActive(selectvideoversion)">VideoSelectLayout</include>
```
Has `selectvideoextra` window ID been removed entirely? I haven't had a chance to test the very most recent changes yet but I thought it still existed from info dialog -- or does extras only use manage dialog now?

If it is still a valid window ID then need to fix up one of the conditions here -- currently the conditions mean no layout at all for `selectvideoextra`

Hitcher replied:
> Has `selectvideoextra` window ID been removed entirely?

No, it's still there.

jjd-uk replied:
It's still there https://github.com/xbmc/xbmc/blob/7413267da70dbb26dcdfbe5e86244343c3716de7/xbmc/guilib/WindowIDs.dox#L137 but it's no longer used. For v21 the only way to access Extras is now via the Manage dialog (DialogVideoManager). Extras will be revisited in v22 so not sure if that's why it remains as a Window ID. Perhaps @ksooo can say if it's being kept to save having to readd in v22 for whatever is planned.

jurialmunkey reviewed (commented):
A couple of  (very) small comments but otherwise looking good.

Hitcher commented:
> Also though, I don't think there's any real downside of simply having a blank list for id=3

I had to do the same.

CrystalP commented:
> DialogSelect sets defaultcontrol to be id 3

With my limited understanding, it seems that changing the C++ to default id 6 (CONTROL_DETAILED_LIST) may work for this dialog, but will break the other CGUIDialogSelect invoked without a SetUseDetails(true), that rely on a default of false (CONTROL_SIMPLE_LIST = 3).

jjd-uk commented:
@CrystalP You can set the default control from the skin side. However I couldn't be 100% sure if changing it to 6 would break anything in Estuary. The blank id3 list does no harm so will probably leave it as it is.

MikeSiLVO commented:
Apologies, I don’t want to intrude on the convo but I currently put both lists for DialogSelect in a grouplist and the focus is on the grouplist id instead of id”3” or id”6”. Seems to work so far but not sure if that could be an issue or not with all these changes. I don’t think it needs to be more complicated than that. Core seems to provide the correct focus.

Using includes for tiny differences, imho, could be the wrong approach but I do trust the process.

Am looking forward to how this turns out. I got some local changes ready to go for when the version/extras gets finalized from the beta 2 differences.

Separate topic but my thanks for all those working on this. I think the latest updates and discussions are headed in the right direction. Some previous critiques of this feature were dismissed immediately by the original creator of the versions PR but imho the direction things have taken since then seem to be major improvements. :)

jurialmunkey commented:
@MikeSiLVO - If you've got both id=3 and id=6 and not doing any special handling for versions selector then yes core sorts everything out for defaultfocus bouncing from 3 to 6. 

The issue above was that we've got a separate include to use a different layout for the versions selector dialog. That separate layout doesn't have an id=3 list because versions selector only uses id=6 and there's no point writing skin code for a list that is never used. Problem was that without id=3 there's then nothing for defaultcontrol to focus.

As mentioned above, no real harm in simply adding a blank id=3 list control though -- and that's the simplest solution if using a separate include for versions layout
```
<control type="list" id="3">
</control>
```

MikeSiLVO commented:
> @MikeSiLVO - If you've got both id=3 and id=6 and not doing any special handling for versions selector then yes core sorts everything out for defaultfocus bouncing from 3 to 6.
> 
> The issue above was that we've got a separate include to use a different layout for the versions selector dialog. That separate layout doesn't have an id=3 list because versions selector only uses id=6 and there's no point writing skin code for a list that is never used. Problem was that without id=3 there's then nothing for defaultcontrol to focus.
> 
> As mentioned above, no real harm in simply adding a blank id=3 list control though -- and that's the simplest solution if using a separate include for versions layout
> 
> ```
> <control type="list" id="3">
> </control>
> ```

Jurialmunkey, you are a much better skinner than I, by a significantly large margin. 
It is almost comically ridiculous that I often feel like a doofus in comparison :anguished:

I was just mentioning my, possibly way wrong opinion, on excluding a list just for removing a few lines that don't even seem to be populated. I do agree that using includes to remove extraneous controls is best. However, taking that approach too far can introduce unwanted side effects. 

Not sure if I am making any sense here so please let me know if I am not understanding.

Separate topic altogether but I also think the textbox addition suggested by Hitcher for the empty list looks rather nice...
Sorry for the noise :)

garbear commented:
I haven't followed the whole convo, but keep in mind that the dialog-select-includes approach has one drawback: The default control is defined before the `<includes>` statement:

```xml
<defaultcontrol always="true">3</defaultcontrol>
```

There is no way to override this, so if no control with ID 3 exists, then focus may not work on dialog open. See how all the game selection dialogs have a control with ID 3 that is intended to be focused by default.

jurialmunkey commented:
> There is no way to override this, so if no control with ID 3 exists, then focus may not work on dialog open. See how all the game selection dialogs have a control with ID 3 that is intended to be focused by default.

@garbear - Well you could conditionally include the defaultcontrol line itself, which is what I was suggesting in this comment: https://github.com/xbmc/xbmc/pull/24286#issuecomment-1872626480

But the simpler approach is having a dummy id=3 control as you mention (and what we eventually settled on here).

garbear commented:
Thanks for the link, ignore what I said.

jjd-uk commented:
Updated to reflect the focus highlight change in https://github.com/xbmc/xbmc/pull/24323

Minors from @jurialmunkey also addressed.

Hitcher commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -676,12 +677,109 @@
+		<value>$INFO[ListItem.Icon]</value>			
+	</variable>
+	<variable name="VideoListThumbVar">
+		<value condition="!String.IsEmpty(Listitem.Art(landscape))">$INFO[Listitem.Art(landscape)]</value>
```
No `ListItem.Art(poster)` here?

jjd-uk replied:
Only wanted 16:9 or 4:3 dimensioned art here.

Hitcher commented on addons/skin.estuary/xml/DialogVideoManager.xml:
```diff
@@ -5,83 +5,81 @@
+				<align>right</align>
+				<aligny>bottom</aligny>
+				<textcolor>grey</textcolor>
+				<label>$INFO[Container(50).ListItem.FileNameAndPath]</label>
```
Probably needs a `<scroll>true</scroll>` for this label.

jjd-uk replied:
Didn't want a scroll as the right alignment means last part of path is always seen.

Hitcher reviewed (approved):
Looks good to me.

JDRIVO commented:
Nice, the only tweak I would make would be to the side button order.

<img width="645" alt="tweak" src="https://github.com/xbmc/xbmc/assets/41508421/b17d9e83-15db-47f8-9d80-ef92d40740f1">

JDRIVO commented:
Not sure if this behaviour currently exists as I haven't tested it, but if it doesn't. What do you fellas think about returning the user to the last clicked side button after the user clicks the right button? So if a user wants to add art to multiple items for instance, they don't have to navigate the list of buttons each time they want to add art.

Hitcher commented:
> So if a user wants to add art to multiple items for instance, they don't have to navigate the list of buttons each time they want to add art.

After adding art it returns to the 'Choose art' button.

jjd-uk committed:
[Estuary] Video Versions Dialog Redesign
